File: backend/src/main/java/com/backend/domain/evaluation/service/AiService.java
package com.backend.domain.evaluation.service;

import com.backend.domain.evaluation.dto.AiDto;
import com.backend.domain.evaluation.service.AiGateway;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AiService {

    private final AiGateway aiGateway;

    public AiDto.CompleteResponse complete(AiDto.CompleteRequest req) {
        String result = aiGateway.complete(req.content(), req.prompt());
        return new AiDto.CompleteResponse(result);
    }
}

File: backend/src/main/java/com/backend/domain/evaluation/service/AiService.kt
package com.backend.domain.evaluation.service

import com.backend.domain.evaluation.dto.AiDto
import org.springframework.stereotype.Service

@Service
class AiService(
    private val aiGateway: AiGateway,
) {

    fun complete(req: AiDto.CompleteRequest): AiDto.CompleteResponse {
        val result = aiGateway.complete(req.content, req.prompt)
        return AiDto.CompleteResponse(result)
    }
}
